dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala

```diff
@@ -64,40 +66,43 @@ case class Table(
   def put(hashPK: Any, attributes: SimplePk*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, attributes: _*)
   def put(hashPK: Any, rangePK: Any, attributes: SimplePk*)(implicit dynamoDB: DynamoDB): Unit = putItem(hashPK, rangePK, attributes: _*)
 
-  def putItem[E <: AnyRef](entity: E)(implicit dynamoDB: DynamoDB): Unit = {
-    val attrs = getAttrValuesToMap(entity, true)
+  def putItem[T: u.TypeTag](entity:T)(implicit dynamoDB: DynamoDB): Unit = {
+    val annotations = getterAnnotationsFromEntity(entity)
```

Comment:
Can you avoid using tuples for entries in the seq? Also, instead of accessing `._1` etc, using pattern matching should make your code more readable. 

For instance, the following code ...

```scala
    val faJoined = annotations.map(tup1 => fields.filter(tup2 => tup2._1 == tup1._1.toString)
      .map(tup2 => (tup1._1.toString, tup1._2.head.toString, {
        tup1._3.toString match {
          case "Int" => tup2._2.asInstanceOf[Int]
          case "String" => tup2._2.asInstanceOf[String]
        }
      }, tup1._3.toString)))
```

can be like this:

```scala
    val faJoined2: Seq[List[(String, String, Any, String)]] = annotations.map {
      case (name, annotations, sig) =>
        val found: Option[(String, AnyRef)] = fields.find { case (fieldName, _) => fieldName == name.toString }
        found match {
          case Some((_, getterValue)) =>
            List((
              name.toString,
              annotations.head.toString,
              {
                sig.toString match {
                  case "Int" => getterValue.asInstanceOf[Int]
                  case "String" => getterValue.asInstanceOf[String]
                }
              }, sig.toString))
          case None =>
            Nil
        }
      case _ => Nil
    }
```

The code is a bit longer but having explicit types and using pattern matching should make it easier-to-understand for everyone (at least for me). Does the `faJoined` really need to be nested seq here? Also, the code calls toString so many times. Probably, `getterAnnotationsFromEntity` can return more usable values.

Reply:
thanks for the recommendations and advice! My code was updated. Let me know if any other changes are needed.